elastic/README.md

```diff
@@ -140,19 +140,96 @@ _Available for Agent versions >6.0_
+To configure this check for an Agent running on a container:
 
 ##### Metric collection
 
-| Parameter            | Value                              |
-| -------------------- | ---------------------------------- |
-| `<INTEGRATION_NAME>` | `elastic`                          |
-| `<INIT_CONFIG>`      | blank or `{}`                      |
-| `<INSTANCE_CONFIG>`  | `{"url": "https://%%host%%:9200"}` |
+Set [Autodiscovery Integrations Templates][22] as Docker labels on your application container:
+
+```yaml
+LABEL "com.datadoghq.ad.check_names"='["elastic"]'
+LABEL "com.datadoghq.ad.init_configs"='[{}]'
+LABEL "com.datadoghq.ad.instances"='[{"url": "https://%%host%%:9200"}]'
+```
+
+##### Log collection
+
+
+Collecting logs is disabled by default in the Datadog Agent. To enable it, see the [Docker log collection documentation][23].
+
+Then, set [Log Integrations][24] as Docker labels:
+
+```yaml
+LABEL "com.datadoghq.ad.logs"='[{"source":"elasticsearch","service":"<SERVICE_NAME>"}]'
+```
+
+##### Trace collection
+
+APM for containerized apps is supported on Agent v6+ but requires extra configuration to begin collecting traces.
+
+Required environment variables on the Agent container:
+
+| Parameter            | Value                                                                      |
+| -------------------- | -------------------------------------------------------------------------- |
+| `<DD_API_KEY>` | `api_key`                                                                  |
+| `<DD_APM_ENABLED>`      | true                                                              |
+| `<DD_APM_NON_LOCAL_TRAFFIC>`  | true |
+
+See [Tracing Kubernetes Applications][17] and the [Kubernetes Daemon Setup][18] for a complete list of available environment variables and configuration.
+
+Then, [instrument your application container][7] and set `DD_AGENT_HOST` to the name of your Agent container.
+
+
+<!-- xxz tab xxx -->
+<!-- xxx tab "Kubernetes" xxx -->
+
+#### Kubernetes
+
+To configure this check for an Agent running on Kubernetes:
+
+##### Metric collection
+
+Set [Autodiscovery Integrations Templates][25] as pod annotations on your application container. Aside from this, templates can also be configure with [a file, a configmap, or a key-value store][26].
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: elasticsearch
+  annotations:
+    ad.datadoghq.com/elasticsearch.check_names: '["elasticsearch"]'
+    ad.datadoghq.com/elasticsearch.init_configs: '[{}]'
+    ad.datadoghq.com/elasticsearch.instances: |
+      [
+        {
+          "url": "https://%%host%%:9200"
+        }
+      ]
+  labels:
+    name: elasticsearch
+```
+
+##### Log collection
+
+
+Collecting logs is disabled by default in the Datadog Agent. To enable it, see the [Kubernetes log collection documentation][27].
+
+Then, set [Log Integrations][28] as pod annotations. This can also be configure with [a file, a configmap, or a key-value store][29].
```

Comment:
```suggestion
Then, set [Log Integrations][28] as pod annotations. This can also be configured with [a file, a configmap, or a key-value store][29].
```